internal-services/catalog/iib-add-fbc-fragment-to-index-image-task.yaml

```diff
@@ -109,9 +113,24 @@ spec:
           fbc_fragment="${3}"
 
           # fetch only builds in progress or completed
-          curl -s "${IIB_SERVICE_URL}/builds?user=${user}&from_index=${from_index}" | \
+          build=$(curl -s "${IIB_SERVICE_URL}/builds?user=${user}&from_index=${from_index}" | \
             jq --arg fbc_fragment "${fbc_fragment}" \
-              '[.items[] |select(.fbc_fragment==$fbc_fragment and .state!="failed")][0] // empty'
+              '[.items[] |select(.fbc_fragment==$fbc_fragment and .state!="failed")][0] // empty')
+
+          if [ "$(jq '.state' < "${build}")" = "complete" ]; then
+            indexImageResolved="$(jq '.index_image_resolved' <  "${build}")"
+            newCatalogCreatedDate="$(date --date "$(skopeo inspect --config "${indexImageResolved}" | \
+              jq -r .created)" "+%s")"
+            upstreamCatalogCreatedDate="$(date --date "$(skopeo inspect --config "$(params.targetIndex)" | \
+              jq -r .created)" "+%s")"
+            # checks if the index_image_resolved in the previous completed build is newer
+            # than the upstream catalog index.
+            # in case the new catalog index is older than the upstream, a new build is
+            # required to assure the catalog integrity.
+            if [ "${newCatalogCreatedDate}" -gt "${upstreamCatalogCreatedDate}" ]; then
+              echo "${build}"
```

Comment:
All we want to do here is echo? The comment makes it sound like we would exit out of the step or something

Reply:
The function was designed (in a previous PR) to output an empty string in case there is no existent build, or the json string of the build in case it can be resumed. 

Considering this, the logic here basically checks if the creationDate of the found image in the build is newer than the one in the published catalog, if so it returns the build, so it can be resumed.

Reply:
Sorry, I didn't expand my UI and realize we were inside a function